### ISSUE TITLE
Timing of rendering?

### ISSUE THREAD

**Issue by EricThomson** [kushalkolar/fastplotlib#66]
Is there a way to time how long initial rendering takes? When I use the usual cell magic like `%%time` it shows how long the Python code takes to run, but it typically takes a while longer before the graphical elements from fastplotlib display. Are there any profiling tools to time these types of things?

**Comment by kushalkolar** [comment]
I think you can do this at the `wgpu` level quite easily, see this for discussion:
https://github.com/pygfx/wgpu-py/issues/256

However I think that in the RFB (remote frame frame buffer) canvas, the fps is currently capped at 30Hz, I haven't dug into where this capping is done.

**Comment by kushalkolar** [comment]
wait I see this is a slightly different question, you're asking how long does it take for the canvas to populate with objects when you call `show()`? This would still really be a wgpu question and interesting to check. For example drawing a single line is basically instant, but drawing 1000 lines may take a second but once the canvas is ready & populated it is very fast to interact with. I think it's related to how in Vulkan objects are "prepared" before being shown?

**Comment by EricThomson** [comment]
Exactly. For instance when you do `%%time` it will often take like a few hundred milliseconds, but when I'm testing rendering a movie with many lines drawn initially over the image, it may take a many seconds before displaying the canvas with the widget (because of the initial population), but as you said once initially rendered, it is very fast.

**Comment by EricThomson** [comment]
I changed title of issue so it is clear I'm not asking how to improve rendering times, but about how to measure times.

**Comment by kushalkolar** [comment]
@almarklein I think rendercanvas takes care of this stuff now right? And profiling the initial render is more within the scope of pygfx and you've kinda done some of this stuff. Feel free to close if that's the case 😄

**Comment by almarklein** [comment]
Well it does measure the time, but this is not yet exposed to the user in any way, so I think the issue is still valid/open.
